src/hotspot/share/prims/universalUpcallHandler.cpp

```diff
@@ -60,6 +60,7 @@ void ProgrammableUpcallHandler::attach_thread_and_do_upcall(jobject rec, address
     should_detach = true;
     thread = Thread::current();
   }
+  MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXWrite, thread));
```

Comment:
(I'm fixing JDK-8262894, somewhat related to this note)

The destructor of this object will run after the thread may detach, so the thread pointer will be invalid at that point. 

It can be fixed probably by
```
{ 
  MACOS_AARCH64_ONLY(ThreadWXEnable wx(WXWrite, thread));
  upcall_helper(thread->as_Java_thread(), rec, buff);
}
```

Or by using Thread::enable_wx instead of RAII style W^X management. 

Sorry, the previous version apparently didn't suffer from this, but now it much more clear what's going on.

Reply:
you are right, I've updated this PR with first approach